pkg/kubectl/cmd/config/create_authinfo.go

```diff
@@ -124,14 +125,14 @@ func (o createAuthInfoOptions) run() error {
 }
 
 // authInfo builds an AuthInfo object from the options
-func (o *createAuthInfoOptions) modifyAuthInfo(existingAuthInfo clientcmdapi.AuthInfo) clientcmdapi.AuthInfo {
+func (o *createAuthInfoOptions) modifyAuthInfo(cmd *cobra.Command, existingAuthInfo clientcmdapi.AuthInfo) clientcmdapi.AuthInfo {
 	modifiedAuthInfo := existingAuthInfo
 
 	var setToken, setBasic bool
 
-	if o.clientCertificate.Provided() {
-		certPath := o.clientCertificate.Value()
-		if o.embedCertData.Value() {
+	if cmd.Flags().Changed(clientcmd.FlagCertFile) {
```

Comment:
I get that you don't love the style. I can make the argument that the old style is tracking the same thing twice. Not a strong argument, I admit. See 'pkg/kubectl/cmd/delete.go' and `pkg/kubectl/cmd/run.go` for others using this style (not my code, don't blame me)